Block Bindings: Remove Block Bindings icon from List View, fixes and CSS updates.

artemiomorales opened the pull request:
## What?
This is a quick follow up to https://github.com/WordPress/gutenberg/pull/59185

- **Removes** the icon from the List View as decided in https://github.com/WordPress/gutenberg/pull/59477#issuecomment-1980371587
- Removes unneeded CSS of the Block Bindings Toolbar indicator.
- Simplifies the SVG of the icon - it uses a single `<path>` now instead of 6 of them.
- [Addresses a console error](https://github.com/WordPress/gutenberg/pull/59185#issuecomment-1973117561). In React, we need to use `fillRule` instead of `fill-rule`.

## Testing Instructions
<!-- Please include step by step instructions on how to test this PR. -->
<!-- 1. Open a post or page. -->
<!-- 2. Insert a heading block. -->
<!-- 3. etc. -->
Please follow the instructions in https://github.com/WordPress/gutenberg/pull/59185

artemiomorales committed:
Fix bindings icon fill color in list view

SantosGuillamot commented:
I'm not sure what is the desired UI here. I can see the paragraph icon, for example, just turns from black to white. Could something like this make sense when it is selected?

![Screenshot 2024-03-01 at 10 04 51](https://github.com/WordPress/gutenberg/assets/34552881/1bbe1779-3166-48ee-aa94-3e1dfc4a533b)

@SaxonF @jasmussen Any thoughts what should be done here?

artemiomorales commented:
> I'm not sure what is the desired UI here. I can see the paragraph icon, for example, just turns from black to white. Could something like this make sense when it is selected?

@SantosGuillamot Hmm that's not right. It should be showing with the purple outline and a white fill:

<img width="1206" alt="Screenshot 2024-02-29 at 2 54 02 PM" src="https://github.com/WordPress/gutenberg/assets/5360536/d78695ee-ca6e-45d1-935c-1f29eb3819b1">

I've tested in a different environment and it's appearing as expected. Are you seeing this in the post editor, site editor, or both? Can you clear your cache and restart your environment to see if the issue persists?

SantosGuillamot commented:
Oh, sorry. I was just suggesting another approach that came to my mind based on the current behavior of other icons. My point was that I'd like to get some feedback to ensure the UI we provide is the desired one.

artemiomorales commented:
> Oh, sorry. I was just suggesting another approach that came to my mind based on the current behavior of other icons. My point was that I'd like to get some feedback to ensure the UI we provide is the desired one.

Ah ok! In that case I'd say that leaving the icon with purple outline and white fill is best, in my opinion.

The blue highlight of the selected block already signals a change, and if we also modify the icon, I believe it introduces cognitive dissonance with too many things changing at once and could look like a mistake with inconsistency of the icon across the UI, though I'd like to hear what design thinks 😄

artemiomorales committed:
Rewrite fill-rule as fillRule to fix console error

jasmussen commented:
Thanks for following up on this. I'd personally love to find some time to explore a couple of more icon variations — the current one is pretty good, but I think we can be even better.

That said, responding to [this comment](https://github.com/WordPress/gutenberg/pull/59477#issuecomment-1972794855) — this is actually what I'd expect: the icon not having color there. So the screenshot there is actually what I'd expect.

michalczaplinski commented:
> That said, responding to https://github.com/WordPress/gutenberg/pull/59477#issuecomment-1972794855 — this is actually what I'd expect: the icon not having color there. So the screenshot there is actually what I'd expect.

This makes sense. So, the final UI should look like: 

<img width="300" alt="connections" src="https://github.com/WordPress/gutenberg/assets/34552881/1bbe1779-3166-48ee-aa94-3e1dfc4a533b">

-----

Additionally, I believe the SVG paths need to be adjusted. Currently the storybook for the connection icon looks like this:

<img width="300" alt="Screenshot 2024-03-04 at 13 04 16" src="https://github.com/WordPress/gutenberg/assets/5417266/b0ea652f-dcc9-4e52-bb31-f408a6d4a886">

michalczaplinski commented:
@jasmussen Could you provide us with an SVG file that meets those requirements and renders correctly in the storybook? The one in the [Figma file](https://www.figma.com/file/W2SO7yJG09O65q7JjfXZ4m/Block-connections?type=design&node-id=385-5610&mode=design&t=eogT3VAAmA76YKM6-0) is not quite correct.

SantosGuillamot commented:
@jasmussen I have just tested the SVG you shared but it seems it is completely white.

If I remove the fill="none" property it looks like this:

![Screenshot 2024-03-05 at 09 39 08](https://github.com/WordPress/gutenberg/assets/34552881/3a066734-8de3-46ed-9420-b3768f21680e)

It might need to fill the whole SVG, right?

SantosGuillamot committed:
Fix connection SVG

SantosGuillamot commented:
I've updated the SVG in [this commit](https://github.com/WordPress/gutenberg/pull/59477/commits/88850c3c504327073dedc84de6beea0471a2e350). This is how it looks in the storybook right now:

![Screenshot 2024-03-05 at 09 44 25](https://github.com/WordPress/gutenberg/assets/34552881/1fac3902-700e-4cab-9f52-90a2d891b54d)

We need to update the CSS logic to match it. Working on it now.

jasmussen commented:
Try this:

```
/**
 * WordPress dependencies
 */
import { SVG, Path } from '@wordpress/primitives';

const connection = (
	<SVG
		viewBox="0 0 24 24"
		xmlns="http://www.w3.org/2000/svg"
		fill-rule="evenodd"
	>
		<Path
			fillRule="evenodd"
			clipRule="evenodd"
			d="M19.53 4.47a.75.75 0 0 1 0 1.06L17.06 8l.77.769a3.155 3.155 0 0 1 .685 3.439 3.15 3.15 0 0 1-.685 1.022v.001L13.23 17.83v.001a3.15 3.15 0 0 1-4.462 0L8 17.06l-2.47 2.47a.75.75 0 0 1-1.06-1.06L6.94 16l-.77-.769a3.154 3.154 0 0 1-.685-3.439 3.15 3.15 0 0 1 .685-1.023l4.599-4.598a3.152 3.152 0 0 1 4.462 0l.769.768 2.47-2.47a.75.75 0 0 1 1.06 0Zm-2.76 7.7L15 13.94 10.06 9l1.771-1.77a1.65 1.65 0 0 1 2.338 0L16.77 9.83a1.649 1.649 0 0 1 0 2.338h-.001ZM13.94 15 9 10.06l-1.77 1.771a1.65 1.65 0 0 0 0 2.338l2.601 2.602a1.649 1.649 0 0 0 2.338 0v-.001L13.94 15Z"
		/>
	</SVG>
);

export default connection;
```

SantosGuillamot commented:
> Try this

I tried something similar [here](https://github.com/WordPress/gutenberg/pull/59477#issuecomment-1978234115) and seems to work. Thanks for your help @jasmussen ☺️

SantosGuillamot committed:
Fix fixRule

SantosGuillamot committed:
Adapt styles to new SVG

SantosGuillamot commented:
Okay, I've adapted the styles and make some changes based on feedback:

### Before SVG and styles changes

* In block toolbar
![Screenshot 2024-03-05 at 10 02 16](https://github.com/WordPress/gutenberg/assets/34552881/a4b641d6-4cc6-4236-bea1-237639a83875)

* In block list when is selected and not selected
![Screenshot 2024-03-05 at 10 02 31](https://github.com/WordPress/gutenberg/assets/34552881/82be60c8-423f-4416-8753-73ec3945660e)

* In storybook
![Screenshot 2024-03-05 at 10 07 15](https://github.com/WordPress/gutenberg/assets/34552881/68ac2320-21b1-45af-914d-4111025a2ef6)

### After SVG and styles changes

* In block toolbar
![Screenshot 2024-03-05 at 09 57 52](https://github.com/WordPress/gutenberg/assets/34552881/e2a1c637-9b9a-4272-ac06-dc0535dd0f7c)


* In block list when is selected and not selected
![Screenshot 2024-03-05 at 09 58 11](https://github.com/WordPress/gutenberg/assets/34552881/a11b09e8-3a3b-4dad-8d63-14b422c5aba1)


* In storybook

![Screenshot 2024-03-05 at 10 08 14](https://github.com/WordPress/gutenberg/assets/34552881/bfd84938-f88e-4aee-90b2-9ac0fd3b0ae0)

---

We can see that, with the new changes, the SVG in the block toolbar looks wider. I know it can be solved with CSS and stroke-width, but I'm not sure if that's the right approach. I'll take a look if it can be solved other way.

Apart from that, I can see that if I enabled the distraction free mode, the icon is not aligned. We must take a look at that as well:

![Screenshot 2024-03-05 at 10 10 18](https://github.com/WordPress/gutenberg/assets/34552881/0054f816-0c82-4649-8214-4f435deb6162)

jasmussen commented:
The icon looks misaligned even in the list view.

I wonder, should we just remove it from the list view for now? There's enough feedback that it isn't clear that's super useful, perhaps this isn't the right approach?

SantosGuillamot committed:
Workaround for distraction free

SantosGuillamot commented:
I added this workaround for the distraction free issue: [link](https://github.com/WordPress/gutenberg/pull/59477/commits/0a157e5972f3fa73bd561711f9f62c9d738a6b7b).

However, I believe we have to explore deeper if we can use core components that already have those styles. For me, ideally, the bindings indicator CSS should be only:

```css
.block-editor-block-bindings-toolbar-indicator svg{
  stroke: var(--wp-bound-block-color);
}
```

And the rest should be managed by core components because it is shared CSS.

SantosGuillamot commented:
> should we just remove it from the list view for now? There's enough feedback that it isn't clear that's super useful, perhaps this isn't the right approach?

I'm fine removing it if the general feedback is that it isn't that useful. We can always add it back later.

michalczaplinski commented:
> For example, if I use ToolbarButton instead of ToolbarItem at [this point](https://github.com/WordPress/gutenberg/blob/0a157e5972f3fa73bd561711f9f62c9d738a6b7b/packages/block-editor/src/components/block-bindings-toolbar-indicator/index.js#L11), everything works out of the box without extra CSS. I believe we decided not to use ToolbarButton because it isn't exactly a button, but we could have something similar I guess. cc: @michalczaplinski

Yes, exactly; we got feedback (https://github.com/WordPress/gutenberg/pull/59185#issuecomment-1959454476) that having a button that does not do anything is not good for a11y. 

I also notice that the icon has a thin black fill:

<img width="289" alt="Screenshot 2024-03-05 at 10 50 42" src="https://github.com/WordPress/gutenberg/assets/5417266/5975df7f-fa46-4639-abf4-9d62f8e6f783">

jasmussen commented:
Tangential question, was the "select parent" button removed?

michalczaplinski commented:
> Tangential question, was the "select parent" button removed?

It wasn't. It will appear before the bound block indicator if the bound block has a parent:

<img width="300" alt="Screenshot 2024-03-05 at 11 58 54" src="https://github.com/WordPress/gutenberg/assets/5417266/e19260d7-fa47-4456-8a23-fbe094f36d53">

jasmussen commented:
> It wasn't. It will appear before the bound block indicator if the bound block has a parent:

Thank you! CC: @annezazu as I think she encountered some glitches.

michalczaplinski committed:
Fix:
- the list view styles,
- the stroke & fill in svg
- remove unnecessary CSS
- lint:fix the scss files

youknowriad commented:
Given that RC1 is today, I'm removing this from 6.5 as non essential.

SantosGuillamot commented:
> Would be great if you could review @SantosGuillamot or @artemiomorales 🙏

I've tested it and it seems to work as expected. However, I'll wait to approve it until we decide if we should remove the icon from the block list or not.

> Given that RC1 is today, I'm removing this from 6.5 as non essential.

Regarding this, right now there is a console error when you select a block that is connected introduced [here](https://github.com/WordPress/gutenberg/pull/59185#issuecomment-1973117561). Should/could it be fixed in a separate PR for 6.5 @youknowriad? Working on it just in case it is possible.

SantosGuillamot commented:
>can we have an isolated PR/issue for it to be added to the sanctioned items on the board.

I have just created a separate PR for that: [link](https://github.com/WordPress/gutenberg/pull/59598). Let me know if that's enough or I should do something else.

youknowriad commented:
That should be fine. Thank you.

youknowriad commented:
Unfortunately no, I have punted several similar small changes as well. It's just that for RC1, it's only critical things that are allowed.

michalczaplinski commented:
> Unfortunately no, I have punted several similar small changes as well. It's just that for RC1, it's only critical things that are allowed.

I understand, thanks for clarifying! 🙂

annezazu commented:
On the parent block front, here's what I saw when testing yesterday:

https://github.com/WordPress/gutenberg/assets/26996883/a93e9b87-d5c6-4fe1-8740-980126e344e9

Note: I'm using WP Playground and have run into tons of issues there so perhaps it's a limitation of playground. Tested again using Playground for this PR and WP nightly:

https://github.com/WordPress/gutenberg/assets/26996883/d84c3005-e3f1-431b-960b-72a9f15fbeab

Another item of feedback shared in the video above-- I'd want the icons in this order: image, block binding, lock. Right now, having the image preview shoved between the two in list view is a bit clunky:

![Screenshot 2024-03-05 at 9 42 58 AM](https://github.com/WordPress/gutenberg/assets/26996883/701854ea-9a34-4663-bf90-da8bf5dad1fa)

artemiomorales committed:
Display parent selector when focused on pattern override

artemiomorales committed:
Move list view connections icon beside lock icon

artemiomorales committed:
Revert "Display parent selector when focused on pattern override"

This reverts commit 717bbe8966962ad3c72687794c9fe8e7f91061f0.

artemiomorales commented:
> On the parent block front, here's what I saw when testing yesterday:

@annezazu This issue with the parent selector not being visible appears to be a pattern overrides issue and I think is best handled separately — pull request created here: https://github.com/WordPress/gutenberg/pull/59621

> Another item of feedback shared in the video above-- I'd want the icons in this order: image, block binding, lock. Right now, having the image preview shoved between the two in list view is a bit clunky

I've swapped the positions of the icons, but I'm not sure the result is any better. Now the connections icon and lock icon appear very far apart due to all of the icons' different sizes and treatment of white space.

Given what @jasmussen mentioned above, perhaps of least resistance is to just remove it for now. Otherwise, Joen do you or perhaps @SaxonF have ideas on how to handle?

<img width="338" alt="Screenshot 2024-03-05 at 9 09 55 PM" src="https://github.com/WordPress/gutenberg/assets/5360536/1c42931e-d8cb-425e-ad8d-11024eebe906">

annezazu commented:
We also can keep it as is if design thinks it's better. I didn't realize how odd the spacing would look in general. Something to follow up on outside of this release and refine.

SantosGuillamot commented:
> Given what @jasmussen mentioned above, perhaps of least resistance is to just remove it for now. Otherwise, Joen do you or perhaps @SaxonF have ideas on how to handle?

In my opinion, we could just remove it until we figure out what is the best strategy for the UX in this regard. It seems it isn't clear if we should have that icon at all.

SaxonF commented:
That sounds fine. We can revisit after 6.5

michalczaplinski committed:
Remove unused CSS styles

michalczaplinski committed:
Remove Icon from List View

michalczaplinski commented on packages/block-editor/src/components/block-bindings-toolbar-indicator/style.scss:
```diff
@@ -1,14 +1,16 @@
-		stroke-width: 1.5;
-		stroke-linecap: round;
-		stroke-linejoin: round;
+	svg {
```
We can now remove those properties because of the updates to the <path> of the SVG.

michalczaplinski commented:
I've removed the Icon from the List View and updated the PR description to reflect that 👍

artemiomorales committed:
Merge branch 'trunk' into fix/block-bindings-indicator-fill

artemiomorales commented:
Most of these are formatting changes. Tested the latest changes and it's looking good to me — the list icon is removed, and the icon still appears in the block toolbar as expected.

michalczaplinski reviewed (approved):
LGTM 👍
